main.nf

```diff
@@ -9,279 +9,812 @@
+
+splitChannel
+   .splitCsv( header: true, sep: '\t' )
+   .set { conditionDeconvolution }
+
+vcfSampleChannel
+   .splitCsv( header: true, sep: '\t' )
+   .map{it ->
+       return it.name
+   }
+   .combine(vcfChannel)
+   .set{ vcfCombineChannel }
+
+/*
+ * STEP 1 - TrimGalore!
+ */
+if (params.skip_trimming) {
+    rawFiles
+       .map{it ->
+        return tuple(it, file(it.reads))
+       }
+       .set{ trimmedFiles }
+    trimgaloreQC = Channel.empty()
+    trimgaloreFastQC = Channel.empty()
+} else {
+  process trim {
+
+       tag { parameters.name }
+
+       input:
+       set val(parameters), file(reads) from rawFiles
+
+       output:
+       set val(parameters), file("TrimGalore/${parameters.name}.fastq.gz") into trimmedFiles
+       file ("TrimGalore/*.txt") into trimgaloreQC
+       file ("TrimGalore/*.{zip,html}") into trimgaloreFastQC
+
+       script:
+       """
+       mkdir -p TrimGalore
+       trim_galore ${reads} --stringency 3 --fastqc --cores ${task.cpus} --output_dir TrimGalore
+       mv TrimGalore/*.fq.gz TrimGalore/${parameters.name}.fastq.gz
+       """
+  }
+}
 
 /*
- * STEP 1 - FastQC
+ * STEP 2 - Map
  */
-process fastqc {
-    tag "$name"
-    publishDir "${params.outdir}/fastqc", mode: 'copy',
-        saveAs: {filename -> filename.indexOf(".zip") > 0 ? "zips/$filename" : "$filename"}
+ process map {
+
+     tag { parameters.name }
+
+     input:
+     set val(parameters), file(fastq) from trimmedFiles
+     each file(fasta) from fastaMapChannel
+
+     output:
+     set val(parameters.name), file("map/*bam") into slamdunkMap
+
+     script:
+     quantseq = params.quantseq ? "-q" : ""
+     endtoend = params.endtoend ? "-e" : ""
+     """
+     slamdunk map -r ${fasta} -o map \
+        -5 ${params.trim5} -n 100 \
+        -a ${params.polyA} -t ${task.cpus} \
+        --sampleName ${parameters.name} \
+        --sampleType ${parameters.type} \
+        --sampleTime ${parameters.time} --skip-sam \
+        ${quantseq} ${endtoend} \
+        ${fastq}
+     """
+ }
+
+ /*
+  * STEP 3 - Filter
+  */
+  process filter {
+
+    label 'slamdunk_process'
+
+    publishDir path: "${params.outdir}/slamdunk/bam", mode: 'copy',
+               overwrite: 'true', pattern: "filter/*bam*",
+               saveAs: { filename ->
+                             if (filename.endsWith(".bam")) file(filename).getName()
+                             else if (filename.endsWith(".bai")) file(filename).getName()
+                       }
+
+      tag { name }
+
+      input:
+      set val(name), file(map) from slamdunkMap
+      each file(bed) from utrFilterChannel
+
+      output:
+      set val(name), file("filter/*bam*") into slamdunkFilter,
+                                               slamdunkCount,
+                                               slamdunkFilterSummary
+
+      script:
+      multimappers = params.multimappers ? "-b ${bed}" : ""
+
+      """
+      slamdunk filter -o filter \
+         ${multimappers} \
```

Comment:
I usually put optional things on the same line like this - otherwise if not specified then the command has a random blank line halfway through which looks strange. Doesn't make any functional difference, but does trigger suspicion when bug-testing.

```suggestion
      slamdunk filter -o filter ${multimappers} \
```

Reply:
Changed.